nimble/transport/socket/src/ble_hci_socket.c

```diff
@@ -743,26 +749,27 @@ static void
 ble_hci_sock_ack_handler(void *arg)
 {
     while (1) {
-        os_eventq_run(&ble_hci_sock_state.evq);
+        ble_npl_eventq_run(&ble_hci_sock_state.evq);
     }
 }
 
 static void
 ble_hci_sock_init_task(void)
 {
-    os_stack_t *pstack;
+    ble_npl_stack_t *pstack;
 
-    pstack = malloc(sizeof(os_stack_t)*BLE_SOCK_STACK_SIZE);
+    pstack = malloc(sizeof(ble_npl_stack_t)*BLE_SOCK_STACK_SIZE);
     assert(pstack);
 
-    os_task_init(&ble_sock_task, "hci_sock", ble_hci_sock_ack_handler, NULL,
-                 MYNEWT_VAL(BLE_SOCK_TASK_PRIO), OS_WAIT_FOREVER, pstack,
-                 BLE_SOCK_STACK_SIZE);
-
-    os_eventq_init(&ble_hci_sock_state.evq);
-    os_callout_stop(&ble_hci_sock_state.timer);
-    os_callout_init(&ble_hci_sock_state.timer, &ble_hci_sock_state.evq,
+    ble_npl_eventq_init(&ble_hci_sock_state.evq);
+    ble_npl_callout_stop(&ble_hci_sock_state.timer);
+    ble_npl_callout_init(&ble_hci_sock_state.timer, &ble_hci_sock_state.evq,
                     ble_hci_sock_rx_ev, NULL);
+
+    ble_npl_task_init(&ble_sock_task, "hci_sock", ble_hci_sock_ack_handler, NULL,
+                      MYNEWT_VAL(BLE_SOCK_TASK_PRIO), OS_WAIT_FOREVER, pstack,
+                      BLE_SOCK_STACK_SIZE);
```

Comment:
this should be Mynewt call with `#ifdef MYNEWT` as in other components

Reply:
But that will fail when compiled for Linux...  Abstracting this call is exactly why NPL exists!